core/chains/evm/logpoller/log_poller.go

```diff
@@ -807,13 +827,11 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 
 		gethLogs, err := lp.ec.FilterLogs(ctx, lp.Filter(big.NewInt(from), big.NewInt(to), nil))
 		if err != nil {
-			var rpcErr client.JsonError
-			if pkgerrors.As(err, &rpcErr) {
-				if rpcErr.Code != jsonRpcLimitExceeded {
-					lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
-					return err
-				}
+			if !isRequestTooLargeError(err) {
+				lp.lggr.Errorw("Unable to query for logs", "err", err, "from", from, "to", to)
+				return err
 			}
+
```

Comment:
I don't think the logic below makes sense in the case of `LimitExceeded`. Why group them together?

Reply:
This logic was designed specifically for the `LimitExceeded` error, which is returned whenever the number of results that would have come back from an `eth_getLogs` request exceeds a limit set by the rpc server.  (In my testing, this was found to be 10,000 for most 3rd party rpc servers, such was alchemy and infura.)

That one has been in the PR since Dec, I just added this other one (`ResultsTooLarge`) because the name of it sounds like it might get returned by some rpc servers under similar circumstances--we have no actual evidence that this is the case, I'm just guessing on that because the name sounds like what `LimitExceeded` does in practice. We can leave this one out, but we can't leave out `LimitExceeded` since that's what this error handling was added for.

Reply:
Is the current comment for `LimitExceeded` accurate? Rate limiting is significantly different from result size limiting.

Reply:
It had been a while since I tested this, so I just went through and tested the responses using curl on a dozen or more different rpc servers.
Sadly, they are much more varied than I'd remembered (although in Dec I think I'd only tried it on 3 providers)--we may have to rely primarily on string parsing 😩, or at least use that in conjunction with error codes.

Here is the response infura gives, which is the LimitExceeded error:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32005,"data":{"from":"0xCB3D","limit":10000,"to":"0x7B737"},"message":"query returned more than 10000 results. Try with this block range [0xCB3D, 0x7B737]."}}
```
Looking at their docs, they also return the same error code if you've exceeded your quota for the day, or are sending requests too fast.  So the comment is accurate, but you're right that they are lumping very different types of errors together under the same error code--I guess we would need to check either "data" or "message" to differentiate it from other types of rate limiting.

Some other examples of responses (I was either mistaken about Alchemy, or they changed it since then).

Alchemy:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32602,"message":"Log response size exceeded. You can make eth_getLogs requests with up to a 2K block range and no limit on the response size, or you can request any block range with a cap of 10K logs in the response. Based on your parameters and the response size limit, this block range should work: [0x0, 0x133e71]"}}
```
(from their docs, -32602 means: "Bad Request: The request is invalid")

LinkPool, Blockdaemon, Chainstack:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32002,"message":"request timed out"}}
```

Quicknode BSC, Fantom:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32614,"message":"eth_getLogs is limited to a 10,000 range"}}
```
(their docs list a number of custom error codes, up to and including -32613 but don't mention -32614 !)

SimplyVC Fanton:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"too wide blocks range, the limit is 100"}}
```

Drpc Avalanche:
```
{"id":1,"jsonrpc":"2.0","error":{"message":"requested too many blocks from 0 to 16777216, maximum is set to 2048","code":-32000}}
```

Nirvana Labs:
(after pause for 60+ seconds... )
```
<!DOCTYPE html>
<html>
  <head>
    <title>503 Backend fetch failed</title>
  </head>
  <body>
    <h1>Error 503 Backend fetch failed</h1>
    <p>Backend fetch failed</p>
    <h3>Guru Meditation:</h3>
    <p>XID: 343710611</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```
(I double checked that the Nirvana Labs endpoint is valid, and responds to an eth_getLogs request with a smaller block range properly--very bad behavior on their part!)